tests/unit/packages.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { expect } from 'chai';
+import * as fs from '../../src/api/fs';
+import * as auth from '../../src/api/auth';
+import { getConfig } from '../../src/api/utils';
+import { resolve } from 'path';
+
+let testPath = resolve(__dirname, 'test.json');
+let testAuth = {
+  organizations: [
+    {
+      name: 'bleenco',
+      teams: [ { name: 'developers', members: [ 'admin' ] } ],
+      members: [ { name: 'admin', role: 'owner' } ],
+      packages: []
+    }
+  ],
+  users: [
+    { name: 'admin', password: 12345, fullName: '' }
+  ],
+  packages: []
+};
+let config = getConfig();
+
+describe('User roles, affiliation and permission specific tests', () => {
+
+  beforeEach(() => {
+    return fs.writeJsonFile(testPath, testAuth);
+  });
+
+  afterEach(() => {
+    return fs.removeFile(testPath);
+  });
+
+  it('should publish a new private package with correct data', () => {
+    let pkgName = 'testPackage';
+    let username = 'admin';
+    let organization = 'bleenco';
+    let version = '0.1.2';
+
+    return auth.publishPackage(pkgName, username, organization, version, testAuth)
+      .then(authentication => {
+        return fs.writeJsonFile(testPath, authentication).then(() => {
+          return fs.readJsonFile(testPath).then(content => {
+            expect(content.organizations[0].packages).to.deep.include({
+              name: pkgName,
+              version: version,
+              teamPermissions: [],
+              memberPermissions: [{ name: username, role: 'owner' }],
+              owner: username
+            });
+          });
+        });
+      });
+  });
+
+  it('should publish a new public package with correct data', () => {
+    let pkgName = 'testPackage';
+    let username = 'admin';
+    let organization = null;
+    let version = '0.1.2';
+
+    return auth.publishPackage(pkgName, username, organization, version, testAuth)
+      .then(authentication => {
+        return fs.writeJsonFile(testPath, authentication).then(() => {
+          return fs.readJsonFile(testPath).then(content => {
+            expect(content.packages).to.deep.include({
```

Comment:
I don't think that's true. `include` also check object values. `equal` is not good for comparing objects, because it expects that the same object will be returned ( an object with same properties and values is not enough).